x/pricefeed/types/market.go

```diff
@@ -72,9 +114,44 @@ func NewPostedPrice(marketID string, oracle sdk.AccAddress, price sdk.Dec, expir
 	}
 }
 
+// Validate performs a basic check of a PostedPrice params.
+func (pp PostedPrice) Validate() error {
+	if strings.TrimSpace(pp.MarketID) == "" {
+		return errors.New("market id cannot be blank")
+	}
+	if pp.OracleAddress.Empty() {
+		return errors.New("oracle address cannot be empty")
+	}
+	if pp.Price.IsNegative() {
+		return fmt.Errorf("posted price cannot be negative %s", pp.Price)
+	}
+	if pp.Expiry.IsZero() {
+		return errors.New("expiry time cannot be zero")
+	}
+	return nil
+}
+
 // PostedPrices type for an array of PostedPrice
 type PostedPrices []PostedPrice
 
+// Validate checks if all the posted prices are valid and there are no duplicated
+// entries.
+func (pps PostedPrices) Validate() error {
+	seenPrices := make(map[string]bool)
+	for _, pp := range pps {
+		if seenPrices[pp.MarketID] {
```

Comment:
This should check if the `marketID` and `OracleAddress` is duplicated. It's fine (and expected) to have multiple posted prices when you export genesis (1 from each oracle for each market).

Reply:
something like this?

changing `seenPrices := make(map[string]string)` and then 
```go
if oracleAddr, ok := seenPrices[pp.MarketID]l ok && oracleAddr == pp.OracleAddress.String()
```